okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -1080,16 +1081,23 @@ public void settings(boolean clearPrevious, Settings settings) {
         if (OkHttpSettingsUtil.isSet(settings, OkHttpSettingsUtil.INITIAL_WINDOW_SIZE)) {
           int initialWindowSize = OkHttpSettingsUtil.get(
               settings, OkHttpSettingsUtil.INITIAL_WINDOW_SIZE);
-          outboundFlow.initialOutboundWindowSize(initialWindowSize);
+          outboundWindowSizeIncreased = outboundFlow.initialOutboundWindowSize(initialWindowSize);
```

Comment:
Hmm... We're updating the window size here. It seems like it would be possible [for a stream to write more DATA](https://github.com/grpc/grpc-java/blob/v1.14.0/okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java#L373) between when this lock is released and the `ackSettings()`.

It looks like [elsewhere](https://github.com/grpc/grpc-java/blob/v1.14.0/okhttp/src/main/java/io/grpc/okhttp/OutboundFlowController.java#L122) we're calling `frameWriter` while the lock is held. It seems we should probably do the same here.